cmd/infra/azure/create.go

```diff
@@ -124,6 +126,13 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 		BaseDomain: o.BaseDomain,
 	}
 
+	if o.ResourceGroupTags != nil {
+		_, err := util.ParseTags(o.ResourceGroupTags)
```

Comment:
why are you calling `ParseTags` here and ignoring the result, then parsing again later?
https://github.com/openshift/hypershift/pull/3490/files#diff-1271b61ceb71b31f782dafb0ecdd632159a5a3d44bb29cd526e05a8f58da9390R267-R270

Reply:
Here we are just checking that the tags are specified correctly to avoid erroring after creating some resources. We can't use it later because the output we need is map[string]*string but the parse function returns map[string]string

Reply:
actually we don't even need the `ParseTags` func even for AWS, we can simply use `cmd.Flags().StringToStringVarP` which will return a map directly

Reply:
hmm cmd.Flags().StringToStringVarP doesnt have much validation we would need to create a new function to validate its input further.  

I think we are better of keeping the aws tags function as it has better validation.  Also we if we use cmd.Flags().StringToStringVarP we would still need to do some conversion to get the right map type.

Reply:
how do you mean it doesn't have much validation? what extra validation do we have in `ParseTags()`?

Reply:
cmd.Flags().StringToStringVarP doesn't fail with empty values or multiple values per key

Reply:
why do we want to fail on empty values? both AWS and Azure tags allow empty values.